src/physics/utils/tropopause_climo_read.F90

```diff
@@ -0,0 +1,241 @@
+    use physconst,            only: pi
+    use time_manager,         only: get_calday
+    use ioFileMod,            only: cam_get_file
+    use cam_pio_utils,        only: cam_pio_openfile
+    use pio,                  only: file_desc_t, var_desc_t, pio_inq_dimid, pio_inq_dimlen
+    use pio,                  only: pio_inq_varid, pio_get_var, pio_closefile, pio_nowrite
+    use phys_vars_init_check, only: mark_as_initialized
+
+    !------------------------------------------------------------------
+    ! ... local variables
+    !------------------------------------------------------------------
+    integer :: i, j, n
+    integer :: ierr
+    type(file_desc_t) :: pio_id
+    integer :: dimid
+    type(var_desc_t) :: vid
+    integer :: nlon, nlat, ntimes
+    integer :: start(3)
+    integer :: count(3)
+
+    ! YMD (01-16, 02-14, ...) corresponding to the time slices of the climatological
+    ! tropopause dataset. Will be converted to day-of-year and stored in tropp_days.
+    integer, parameter :: dates(tropp_slices) = (/ 116, 214, 316, 415, 516, 615, &
+         716, 816, 915, 1016, 1115, 1216 /)
+    type(interp_type) :: lon_wgts, lat_wgts
+    real(kind_phys), allocatable :: tropp_p_in(:,:,:)
+    real(kind_phys), allocatable :: lat(:)
+    real(kind_phys), allocatable :: lon(:)
+    real(kind_phys) :: to_lats(pcols), to_lons(pcols)
+    real(kind_phys), parameter :: d2r=pi/180._kind_phys, zero=0._kind_phys, twopi=pi*2._kind_phys
+    character(len=shr_kind_cl) :: locfn
+    character(len=shr_kind_cm) :: errmsg
+
+    errmsg = ''
+
+    !-----------------------------------------------------------------------
+    !       ... open netcdf file
+    !-----------------------------------------------------------------------
+    call cam_get_file (tropopause_climo_file, locfn, allow_fail=.false.)
+    call cam_pio_openfile(pio_id, trim(locfn), PIO_NOWRITE)
+
+    !-----------------------------------------------------------------------
+    !       ... get time dimension
+    !-----------------------------------------------------------------------
+    ierr = pio_inq_dimid( pio_id, 'time', dimid )
+    ierr = pio_inq_dimlen( pio_id, dimid, ntimes )
+    if( ntimes /= tropp_slices ) then
+       write(iulog,*) 'tropopause_climo_read_file: number of months = ',ntimes,'; expecting ',tropp_slices
+       call endrun('tropopause_climo_read_file: number of months in file incorrect')
+    end if
+    !-----------------------------------------------------------------------
+    !       ... get latitudes
+    !-----------------------------------------------------------------------
+    ierr = pio_inq_dimid( pio_id, 'lat', dimid )
+    ierr = pio_inq_dimlen( pio_id, dimid, nlat )
+    allocate( lat(nlat), stat=ierr, errmsg=errmsg )
+    if( ierr /= 0 ) then
+       write(iulog,*) 'tropopause_climo_read_file: lat allocation error = ',ierr
+       call endrun('tropopause_climo_read_file: failed to allocate lat, error = ' // errmsg)
+    end if
```

Comment:
this isn't critical, but for each of these allocation errors, you could use `check_allocate` instead (in cam_abortutils). Might align it more with existing CAM-SIMA code

Reply:
Thanks @peverwhee! Looks like I can't pass the `errmsg` suggested by @nusbaume into `check_allocate` (it seems to just take the error code `ierr`). Maybe I could open an issue to add a message argument to `check_allocate` and pass in the `errmsg` returned by `allocate` into it in the future?